cloudSetup/cloudSetup.tf

```diff
@@ -0,0 +1,182 @@
+# Keys generated without the -m PEM will lead to login fails.
+
+provider "aws" {
+  region = "eu-central-1"
+  profile = "c1"
+}
+
+resource "aws_vpc" "vpc" {
+  cidr_block = "10.0.0.0/16"
+  tags = {
+    "exa:owner" : var.owner,
+    "exa:deputy" : var.deputy
+    "exa:project" : var.project,
+    "exa:project.name" : var.project_name
+    "exa:stage" : var.stage
+    "Name" : "VPC for ${var.project}"
+  }
+}
+
+resource "aws_subnet" "subnet" {
+  vpc_id = aws_vpc.vpc.id
+  cidr_block = "10.0.0.0/24"
+
+  tags = {
+    "exa:owner" : var.owner,
+    "exa:deputy" : var.deputy
+    "exa:project" : var.project
+    "exa:project.name" : var.project_name
+    "exa:stage" : var.stage
+    "Name" : "Subnet for ${var.project}"
+  }
+}
+
+
+resource "aws_default_route_table" "my_routing_table" {
+  default_route_table_id = aws_vpc.vpc.default_route_table_id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.gw.id
+  }
+  tags = {
+    "exa:owner" : var.owner,
+    "exa:deputy" : var.deputy
+    "exa:project" : var.project
+    "exa:project.name" : var.project_name
+    "exa:stage" : var.stage
+    "Name" : "Route Table for ${var.project}"
+  }
+}
+
+resource "aws_security_group" "exasol_db_security_group" {
+  name = "allow_tls"
+  description = "Allow TLS inbound traffic"
+  vpc_id = aws_vpc.vpc.id
+
+  ingress {
+    description = "SSH from the internet"
+    from_port = 22
+    to_port = 22
+    protocol = "tcp"
+    cidr_blocks = [
+      "0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port = 0
+    protocol = "-1"
+    to_port = 0
+    self = true
+  }
+
+  egress {
+    from_port = 0
+    to_port = 0
+    protocol = "-1"
+    cidr_blocks = [
+      "0.0.0.0/0"]
+  }
+
+  tags = {
+    "exa:owner" : var.owner,
+    "exa:deputy" : var.deputy
+    "exa:project" : var.project
+    "exa:project.name" : var.project_name
+    "exa:stage" : var.stage
+    "Name" : "Security Group for Exasol cluster for ${var.project}"
+  }
+}
+
+resource "aws_key_pair" "test_pc_key_pair" {
+  key_name = "exasol-test-setup-abstraction-test-pc-key"
+  public_key = file("~/.ssh/id_rsa.pub")
+}
+
+resource "aws_internet_gateway" "gw" {
+  vpc_id = aws_vpc.vpc.id
+
+  tags = {
+    "exa:owner" : var.owner,
+    "exa:deputy" : var.deputy
+    "exa:project" : var.project
+    "exa:project.name" : var.project_name
+    "exa:stage" : var.stage
+    "Name" : "Gateway for Exasol cluster for ${var.project}"
+  }
+}
+
+resource "random_password" "exasol_sys_password" {
+  length = 16
+  min_upper = 1
+  min_lower = 1
+  min_numeric = 1
+  # with some special chars login does not work
+  special = false
+}
+
+resource "random_password" "exasol_admin_password" {
+  length = 20
+  number = true
+  special = true
+  min_special = 1
+  min_numeric = 1
+}
+
+module "exasol" {
+  source = "../../terraform-aws-exasol"
+  cluster_name = "jakobs-exasol"
+  database_name = "exadb"
+  ami_image_name = "Exasol-R7.0.8-BYOL"
```

Comment:
It would be nice also to accept this as parameter.
We could start the setup for different versions of the Exasol dabase.

Reply:
That would be useful for matrix test right? Since we don't do them right now I think we can keep it just here. In my eyes a variable should be something that's intended to be changed.